pkg/network/common/egressip.go

```diff
@@ -607,7 +804,7 @@ func (eit *EgressIPTracker) allocateExistingEgressIPs(allocation map[string][]st
 				break
 			}
 		}
-		if found && !node.offline {
+		if found && !node.offline && (node.capacity == unlimitedNodeCapacity || node.capacity-len(allocation[node.nodeName]) > 0) {
```

Comment:
No, not quite. `allocation[node.nodeName] > 0` doesn't really make sense because the value of the map `allocation` is a slice of strings of IPs that have been allocated to `node.nodeName`. Hence why `len(allocation[node.nodeName])` is used to count the allocations assigned to that node.